### PR TITLE
Disallow combining of DaskStream and Stream objects

### LINKED ISSUE #172: ENH: warn user if combining dask streamz with not dask streamz
Problems seem to occur if a dask stream are combined with not a dask stream (causing futures to not be evaluated when running functions). It might be nice to warn users if they have subscribed a dask stream to a non dask stream without a scatter/gather.

### PR DESCRIPTION
**Pull request by martindurant**
Fixes #172

### CODE REVIEW

**Comment by martindurant** [comment]
This outright errors, which is the extreme of warning. Not sure if you wanted to go that far @CJ-Wright .

**Comment by CJ-Wright** [comment]
I guess one could come up with a very wonky use case which runs operations on the `Futures` objects themselves which would require hooking up dask/not dask streams but that seems like a very janky use case and maybe we shouldn't support unless people demand it.

File: streamz/core.py [hunk 1]
```diff
@@ -105,11 +105,18 @@ class Stream(object):
 
     def __init__(self, upstream=None, upstreams=None, stream_name=None,
                  loop=None, asynchronous=None, ensure_io_loop=False):
+        from .dask import DaskStream
```
Comment (mrocklin) [inline streamz/core.py]: This creates a dependency on Dask?

File: streamz/core.py [hunk 2]
```diff
@@ -105,11 +105,18 @@ class Stream(object):
...
         self.downstreams = OrderedWeakrefSet()
         if upstreams is not None:
             self.upstreams = list(upstreams)
         else:
             self.upstreams = [upstream]
+        if upstreams:
+            if (all(isinstance(u, DaskStream) for u in upstreams) !=
+                    any(isinstance(u, DaskStream) for u in upstreams)):
+                raise ValueError('Must combine only DaskStreams '
+                                 'or no DaskStreams')
```
Comment (mrocklin) [inline streamz/core.py]: Are there no valid use cases of doing this?

**Comment by martindurant** [comment]
OK, we can close this as overenthusiastic, following comments.

**Pull request closed without merging**